impersonate_login/models/model.py

```diff
@@ -0,0 +1,23 @@
+# Copyright (C) 2024 Akretion (<http://www.akretion.com>).
+# @author Kévin Roche <kevin.roche@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+from odoo.http import request
+
+
+class BaseModel(models.AbstractModel):
+    _inherit = "base"
+
+    @api.model
+    def _create(self, data_list):
+        if request and request.session.impersonate_from_uid:
+            user = self.env["res.users"].browse(request.session.impersonate_from_uid)
+            return super(BaseModel, self.with_user(user))._create(data_list)
+        return super()._create(data_list)
+
+    def write(self, vals):
+        if request and request.session.impersonate_from_uid:
+            user = self.env["res.users"].browse(request.session.impersonate_from_uid)
+            return super(BaseModel, self.with_user(user)).write(vals)
+        return super().write(vals)
```

Comment:
May be I am mistaken but **I have big doubts** about this implementation: what if my module B override create or write (super common) and does something, (potentially even like writing or creating some other record) on the behalf of the current self.env.user before calling super().create or super().write?

 **Because my module B will not depend on the impersonate_login module you _create and write override can be arbitrarily called before or after the ones on B**... So I fear I have no guarantee your impersonate_login module will have impersonated the user yet in my overrides in B...


What about renaming the module `base_schizophrenia` or something like that?

Shouldn't this user switch be done during the login phase in res.users instead or in https://github.com/OCA/server-auth/blob/16.0/auth_admin_passkey/models/res_users.py#L57 ?

cc @sebastienbeau @legalsylvain

Reply:
also I wonder if the repo should not rather be https://github.com/OCA/server-auth like for auth_admin_passkey for instance...